app/src/main/java/com/owncloud/android/ui/activity/ShareActivity.kt

```diff
@@ -0,0 +1,186 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.ui.activity
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import com.nextcloud.client.account.User
+import com.owncloud.android.R
+import com.owncloud.android.databinding.ShareActivityBinding
+import com.owncloud.android.datamodel.SyncedFolderProvider
+import com.owncloud.android.datamodel.ThumbnailsCacheManager
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.files.ReadFileRemoteOperation
+import com.owncloud.android.lib.resources.files.model.RemoteFile
+import com.owncloud.android.lib.resources.shares.ShareType
+import com.owncloud.android.operations.GetSharesForFileOperation
+import com.owncloud.android.ui.fragment.FileDetailSharingFragment
+import com.owncloud.android.ui.fragment.FileDetailsSharingProcessFragment
+import com.owncloud.android.ui.fragment.FileDetailsSharingProcessFragment.Companion.newInstance
+import com.owncloud.android.utils.DisplayUtils
+import com.owncloud.android.utils.MimeTypeUtil
+import javax.inject.Inject
+
+/**
+ * Activity for sharing files.
+ */
+class ShareActivity : FileActivity() {
+
+    @JvmField
+    @Inject
+    var syncedFolderProvider: SyncedFolderProvider? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val binding = ShareActivityBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        val optionalUser = user
+        if (!optionalUser.isPresent) {
+            finish()
+            return
+        }
+
+        setupIcon(binding, optionalUser.get())
+        setupFileInfo(binding)
+        readFile(binding, optionalUser.get())
+
+        if (savedInstanceState == null) {
+            addShareFragment(optionalUser.get())
+        }
+    }
+
+    private fun setupIcon(binding: ShareActivityBinding, user: User) {
+        if (file.isFolder) {
+            val isAutoUploadFolder =
+                SyncedFolderProvider.isAutoUploadFolder(syncedFolderProvider, file, user)
+            val overlayIconId = file.getFileOverlayIconId(isAutoUploadFolder)
+            val drawable = MimeTypeUtil.getFileIcon(preferences.isDarkModeEnabled, overlayIconId, this, viewThemeUtils)
+            binding.shareFileIcon.setImageDrawable(drawable)
+        } else {
+            binding.shareFileIcon.setImageDrawable(
+                MimeTypeUtil.getFileTypeIcon(
+                    file.mimeType,
+                    file.fileName,
+                    this,
+                    viewThemeUtils
+                )
+            )
+            if (MimeTypeUtil.isImage(file)) {
+                val remoteId = file.remoteId.toString()
```

Comment:
```suggestion
                val remoteId = ThumbnailsCacheManager.PREFIX_THUMBNAIL + file.remoteId
```

Previously - before the conversion to Kotlin - the preview image would always fail to load because the key didn't have the appropriate prefix. This should fix that.